src/ClientData/include/ClientData/ScopeIdProvider.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CLIENT_DATA_API_SCOPE_ID_PROVIDER_H_
+#define ORBIT_CLIENT_DATA_API_SCOPE_ID_PROVIDER_H_
+
+#include <cstdint>
+
+#include "ClientData/TimerTrackDataIdManager.h"
+#include "GrpcProtos/capture.pb.h"
+
+namespace orbit_client_data {
+
+// The inferface defines a map from `TimerInfo` to ids. When called twice on identical `TimerInfo`
+// instances, it returns the same ids.
+class ScopeIdProvider {
+ public:
+  virtual ~ScopeIdProvider() = default;
+
+  [[nodiscard]] virtual uint64_t ProvideId(const TimerInfo& timer_info) = 0;
+};
+
+// If `timer_info.function_id` is not invalid, it is returned.
```

Comment:
Personal preference: Again, I would have the comment slightly more describing on a higher level what we expect from that class/function, not soo much how in detail it works.

This class, unless the timer does already have an id, is assigning an id for timers based on the equality of their names. I.e. all timers with the same name will have the same id. Currently this is only implemented for manual instrumentation scopes.